tools/shared.py

```diff
@@ -2874,16 +2883,20 @@ def get_binaryen_bin():
     return os.path.join(BINARYEN_ROOT, 'bin')
 
   @staticmethod
-  def get_binaryen_command(tool, args=[], debug=False):
+  def get_binaryen_command(tool, args=[], infile=None, outfile=None, debug=False):
```

Comment:
Why not make them mandatory?  (and come before the extra args?)

Reply:
The output file should definitely be optional, as sometimes we don't pass it.

The input is currently passed all the time, so I can change that one if you want?

Reply:
Sounds good how about:

```
def get_binaryen_command(tool, infile, outfile=None, args=[], debug=False):
```

Reply:
sounds good @sbc100, I rewrote it that way.